tensorflow/core/kernels/data/dataset_test_base.cc

```diff
@@ -0,0 +1,182 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/kernels/data/dataset_test_base.h"
+
+namespace tensorflow {
+namespace data {
+
+std::unique_ptr<OpKernel> DatasetOpsTestBase::CreateOpKernel(
+    const NodeDef& node_def) {
+  Status status;
+  std::unique_ptr<OpKernel> kernel =
+      tensorflow::CreateOpKernel(device_type_, device_.get(), allocator_,
+                                 node_def, TF_GRAPH_DEF_VERSION, &status);
+  TF_CHECK_OK(status) << status;
+  return kernel;
+}
+
+DatasetBase* DatasetOpsTestBase::CreateDataset(OpKernel* kernel,
+                                               OpKernelContext* context) {
+  TF_CHECK_OK(RunOpKernel(kernel, context));
+  // Assume that DatasetOp has only one output.
+  return GetDatasetFromContext(context, 0);
+}
+
+std::unique_ptr<IteratorBase> DatasetOpsTestBase::CreateIterator(
+    OpKernelContext* context, DatasetBase* dataset,
+    std::unique_ptr<IteratorContext>* iterator_context) {
+  *iterator_context = MakeUnique<IteratorContext>(context);
+  IteratorContext::Params params(iterator_context->get());
+  params.function_handle_cache = new FunctionHandleCache(flr_);
+  iterator_context->reset(new IteratorContext(params));
```

Comment:
you should use `absl::make_unique` here

Also, you do not need to set the `IteratorContext` twice, instead you can do:

```
IteratorContext::params params(context);
... update parameters ...
 *iterator_context = absl::make_unique<IteratorContext>(std::move(params));
```